maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/AbstractGeneratorMojo.java

```diff
@@ -163,7 +164,36 @@
      * @since 3.3
      */
     @Parameter
-    protected List<String> packagingTypes = Arrays.asList( "maven-plugin" );
+    protected List<String> packagingTypes = Collections.singletonList( "maven-plugin" );
```

Comment:
Can this ever change?

Reply:
unsure, I just got rid of warning (one param to Arrays.asList).

Reply:
IF this will never change then make is static final string?

Reply:
in any case, that change is unrelated to this PR... Again, I just fixed a warning, but I can undo this change

Reply:
Yes, you are right. Let's keep your change. I have just noticed this and it felt awkward.